version: '3.8'

services:
  db:
    image: mariadb:10.7.1
    container_name: database_sym6
    restart: on-failure
    ports:
      - '4306:3306'
    environment:
      MARIADB_ROOT_PASSWORD: '${MARIADB_ROOT_PASSWORD}'
      MARIADB_USER: '${MARIADB_USER}'
      MARIADB_PASSWORD: '${MARIADB_PASSWORD}'
      MARIADB_DATABASE: '${MARIADB_DATABASE}'
    volumes:
      - ./docker/db/default.cnf:/etc/mysql/conf.d/my.cnf
      - ./docker/db/data:/var/lib/mysql
    networks:
      - learn-symfony-6-network

  php:
    container_name: php_sym6
    build:
      context: ./docker/php-fpm/
    environment:
      APP_ENV: '${APP_ENV}'
      APP_DEBUG: '${APP_DEBUG}'
    ports:
      - "9000"
    volumes:
      - ./symfony6-app:/var/www/symfony6-app
      - ./symfony6-app/public:/var/www/symfony6-app/public
    depends_on:
      - db
    networks:
      - learn-symfony-6-network

  web-server:
    container_name: web-server-sym6
    build:
      context: ./docker/web-server
      dockerfile: Dockerfile
    restart: on-failure
    ports:
      - "8081:80"
    volumes:
      - ./docker/web-server/default.conf:/etc/nginx/conf.d/default.conf
      - ./symfony6-app/public:/var/www/symfony6-app/public
    depends_on:
      - db
      - php
    links:
      - php
    networks:
      - learn-symfony-6-network

networks:
  learn-symfony-6-network:
